src/charm.py

```diff
@@ -174,6 +175,34 @@ def _on_register_user_action(self, event: ActionEvent) -> None:
         results = {"register-user": True, "user-password": user.password}
         event.set_results(results)
 
+    def _on_change_user_admin_action(self, event: ActionEvent) -> None:
+        """Change user admin and report action result.
+
+        Args:
+            event: Event triggering the reset instance action.
+        """
+        results = {
+            "change-user-admin": False,
+        }
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            event.fail("Failed to connect to container")
+            return
+        try:
+            admin_access_token = secret_storage.get_admin_access_token(self)
```

Comment:
In the current implementation that could return none (and suggested implementation is to throw an exception). This should be handled here